metrics-facade-base/src/main/java/com/ringcentral/platform/metrics/instrument/executors/MonitoredScheduledExecutorService.java

```diff
@@ -0,0 +1,171 @@
+ *       {@code metricKeyProvider.apply(name("idle"))}</li>
+ *   <li>{@link com.ringcentral.platform.metrics.timer.Timer} for task execution time: {@code metricKeyProvider.apply(name("execution"))}</li>
+ * </ul>
+ *
+ * If the underlying (parent) {@link ScheduledExecutorService} is a {@link ForkJoinPool}, it additionally provides the following {@link com.ringcentral.platform.metrics.var.longVar.LongVar} metrics:
+ * <ul>
+ *   <li>{@code metricKeyProvider.apply(name("tasks", "stolen"))}: parent.getStealCount()</li>
+ *   <li>{@code metricKeyProvider.apply(name("tasks", "queued"))}: parent.getQueuedTaskCount()</li>
+ *   <li>{@code metricKeyProvider.apply(name("threads", "active"))}: parent.getActiveThreadCount()</li>
+ *   <li>{@code metricKeyProvider.apply(name("threads", "running"))}: parent.getRunningThreadCount()</li>
+ * </ul>
+ *
+ * To create this wrapper, use the {@link MonitoredScheduledExecutorServiceBuilder}:
+ * <pre>
+ * {@code
+ * // Example 1: Default naming (based in a sequence of positive integer number)
+ * // executor_service_submitted_total{name="1",} 6.0
+ * monitoredScheduledExecutorService(parent, registry).build();
+ *
+ * // Example 2: Custom name
+ * // executor_service_submitted_total{name="customName",} 6.0
+ * monitoredScheduledExecutorService(parent, registry)
+ *   .name("customName")
+ *   .build();
+ *
+ * // Example 3: Exclude name from labels and make it a part of the name
+ * // executor_service_nameNotLabel_submitted_total 6.0
+ * monitoredScheduledExecutorService(parent, registry)
+ *   .name("nameNotLabel")
+ *   .nameAsLabel(false)
+ *   .build();
+ *
+ * // Example 4: Custom name prefix and additional labels
+ * // customMetricNamePrefix_submitted_total{sample="MonitoredExecutorService", name="customMetricNamePrefixAndAdditionalLabels",} 6.0
+ * monitoredScheduledExecutorService(parent, registry)
+ *   .name("customMetricNamePrefixAndAdditionalLabels")
+ *   .metricNamePrefix(name("customMetricNamePrefix"))
+ *   .prefixLabelValues(labelValues(SAMPLE.value("MonitoredExecutorService")))
+ *   .additionalLabelValues(labelValues(SERVICE.value("myService")))
+ *   .build();
+ * }
+ * </pre>
+ *
+ * For more details, see the relevant example in {@code MonitoredScheduledExecutorServiceSample} within the metrics-facade-samples submodule:
+ * <a href="https://github.com/ringcentral/metrics-facade/tree/master/metrics-facade-samples">https://github.com/ringcentral/metrics-facade/tree/master/metrics-facade-samples</a>
+ */
+
+public class MonitoredScheduledExecutorService extends AbstractMonitoredExecutorService<ScheduledExecutorService> implements ScheduledExecutorService {
+
+    private final Rate scheduledOnceRate;
+    private final Rate scheduledRepetitivelyRate;
+    private final Rate scheduledOverrunRate;
+    private final Histogram periodPercentHistogram;
+
+    public MonitoredScheduledExecutorService(
```

Comment:
I suggest to make it `protected` and leave a link to `MonitoredScheduledExecutorServiceBuilder` once more.

Reply:
I don't think we should restrict the user this way, because for some users it may be easier to create an object directly, and this constructor, for example, doesn't restrict the MetricName/Key format. However, I'll add the link.